### PR TITLE
README.md: add more information about Web IDE

### PR DESCRIPTION
**Pull request by gvancuts**
Add more information and pointers on how to get started
with the Web IDE (inc. link to the video tutorial that was
posted on the Zephyr Project website).

Re-organize a bit the content and index to improve the flow
and readibility

Signed-off-by: Geoffroy Van Cutsem <geoffroy.vancutsem@intel.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -705,8 +667,59 @@ normal you can usually fix it by restarting the two scripts and running QEMU
...
+- `deps/` - Contains dependency repos and scripts for working with them.
+- `docs/` - Documentation in Markdown format (use API.md as index).
+- `outdir/` - Directory generated by build, can be safely removed.
+- `samples/` - Sample JavaScript files that can be built with make JS=<path>.
+- `scripts/` - Subdirectory containing tools useful during development.
+- `src/` - JS API bindings for JerryScript written directly on top of Zephyr.
+- `tests/` - JavaScript unit tests (incomplete).
+
+
+[Web IDE]: https://01org.github.io/zephyrjs-ide
```
Comment (grgustaf) [inline README.md]: What did you intend this bit to do? It doesn't seem to show up... other than that +1 with my little fix.

File: README.md [hunk 2]
```diff
@@ -189,8 +189,7 @@ make dfu
 
 This will flash both the images to the device using the dfu-util program.
 
-If you get a permission error, make sure you followed the [#Join the plugdev
-group] instructions above for this user. You shouldn't need to run this command
+If you get a permission error, make sure you followed the [plugdev group](#join-the-plugdev-group) instructions the [plugdev group](#join-the-plugdev-group) instructions above for this user. You shouldn't need to run this command
```
Comment (gvancuts) [inline README.md]: Good catch, thanks for fixing this for me @grgustaf ! The [plugdev group' link appears twice though now. Can you correct this?

File: README.md (hunk 1 above)
Comment (gvancuts) [inline README.md]: This is a mechanism to define a link that can be used multiple times throughout the doc without having to type the actual link again and again. With that definition line (which, as you noted, does not appear in the rendering) you can insert this hyperlink by simply typing [Web IDE] in the raw text.

**Comment by gvancuts** [comment]
- Fixed up @grgustaf addition
- Squashed both into one commit
- Rebased and force-pushed

File: README.md (hunk 1 above)
Comment (grgustaf) [inline README.md]: I have much to learn from your github fu. :)

**Pull request merged**